zeebe/logstreams/src/main/java/io/camunda/zeebe/logstreams/impl/log/SequencerMetrics.java

```diff
@@ -7,58 +7,128 @@
+
+  SequencerMetrics(final MeterRegistry meterRegistry) {
+    Gauge.builder(QUEUE_SIZE.getName(), queueSize, Number::longValue)
+        .description(QUEUE_SIZE.getDescription())
+        .register(meterRegistry);
+
+    batchSize =
+        DistributionSummary.builder(BATCH_SIZE.getName())
+            .description(BATCH_SIZE.getDescription())
+            .serviceLevelObjectives(BATCH_SIZE.getDistributionSLOs())
+            .register(meterRegistry);
+    batchLengthBytes =
+        DistributionSummary.builder(BATCH_LENGTH_BYTES.getName())
+            .description(BATCH_LENGTH_BYTES.getDescription())
+            .serviceLevelObjectives(BATCH_LENGTH_BYTES.getDistributionSLOs())
+            .register(meterRegistry);
   }
 
   void setQueueSize(final int length) {
     queueSize.set(length);
   }
 
   void observeBatchSize(final int size) {
-    batchSize.observe(size);
+    batchSize.record(size);
   }
 
   void observeBatchLengthBytes(final int lengthBytes) {
     final int batchLengthKiloBytes = Math.floorDiv(lengthBytes, 1024);
-    batchLengthBytes.observe(batchLengthKiloBytes);
+    batchLengthBytes.record(batchLengthKiloBytes);
+  }
+
+  @SuppressWarnings("NullableProblems")
+  public enum SequencerMetricsDoc implements ExtendedMeterDocumentation {
+    /** Current length of queue, i.e. how many entry batches are available to the appender */
+    QUEUE_SIZE {
+      @Override
+      public String getDescription() {
+        return "Current length of queue, i.e. how many entry batches are available to the appender";
+      }
+
+      @Override
+      public String getName() {
+        return "zeebe.sequencer.queue.size";
+      }
+
+      @Override
+      public Type getType() {
+        return Type.GAUGE;
+      }
+    },
+
+    /** Histogram over the number of entries in each batch that is appended */
+    BATCH_SIZE {
+
+      public static final double[] BUCKETS = {1, 2, 3, 5, 10, 25, 50, 100, 500, 1000};
+
+      @Override
+      public String getDescription() {
+        return "Histogram over the number of entries in each batch that is appended";
+      }
+
+      @Override
+      public String getName() {
+        return "zeebe.sequencer.batch.size";
+      }
+
+      @Override
+      public Type getType() {
+        return Type.DISTRIBUTION_SUMMARY;
+      }
+
+      @Override
+      public double[] getDistributionSLOs() {
+        return BUCKETS;
+      }
+    },
+
+    /** Histogram over the size, in Kilobytes, of the sequenced batches */
+    BATCH_LENGTH_BYTES {
+
+      public static final double[] BUCKETS = {0.256, 0.512, 1, 4, 8, 32, 128, 512, 1024, 4096};
+
+      @Override
+      public String getDescription() {
+        return "Histogram over the size, in Kilobytes, of the sequenced batches";
+      }
+
+      @Override
+      public String getName() {
+        return "zeebe.sequencer.batch.length.bytes";
+      }
+
+      @Override
+      public String getBaseUnit() {
+        return "bytes";
```

Comment:
:x: is this "KiB"? The buckets are in KiB and the metric is recorded in KiB as well

Reply:
Good point, yes.